src/Articulate/UmbracoEventHandler.cs

```diff
@@ -122,7 +122,7 @@ private static void PageCacheRefresher_CacheUpdated(PageCacheRefresher sender, U
                     if (content.ContentType.Alias.InvariantEquals("Articulate"))
                     {
                         //ensure routes are rebuilt
-                        UmbracoContext.Current.Application.ApplicationCache.RequestCache.GetCacheItem("articulate-refresh-routes", () => true);
+                        ApplicationContext.Current.ApplicationCache.RequestCache.GetCacheItem("articulate-refresh-routes", () => true);
```

Comment:
This is acceptable use of singletons since there's no other option here and ApplicationContext.Current is much better than UmbracoContext.Current.Application, good catch.